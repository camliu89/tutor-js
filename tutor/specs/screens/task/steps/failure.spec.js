import Failure from '../../../../src/screens/task/step/failure';
import Task from '../../../../src/models/student-tasks/task';
import Step from '../../../../src/models/student-tasks/step';
import Raven from '../../../../src/models/app/raven';
jest.mock('../../../../src/models/app/raven');


describe('Task Failure', () => {
  let props;

  beforeEach(() => {
    props = {
      task: new Task(),
      step: new Step({ id: 1234 }),
    };
  });

  it('matches snapshot', () => {
    expect(<Failure {...props} />).toMatchSnapshot();
  });

  it('logs step message', () => {
    const fail = mount(<Failure {...props} />);
    expect(Raven.log).toHaveBeenCalledWith(
      expect.stringContaining('Failed to save assignment step id: 1234')
    );
    fail.unmount();
  });

  it('logs task message', () => {
    props.task.api.errors = { foo: 'bar', last: { config: { method: 'get' } } };
    const fail = mount(<Failure {...props} />);

    expect(Raven.log).toHaveBeenCalledWith(
      expect.stringContaining('Failed to load assignment, errors: Foo bar')
    );
    fail.unmount();
  });

});
