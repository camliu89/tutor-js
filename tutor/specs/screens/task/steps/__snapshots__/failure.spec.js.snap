// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Failure matches snapshot 1`] = `
<Failure
  step={
    StudentTaskStep {
      "answer_id": undefined,
      "content": undefined,
      "correct_answer_id": undefined,
      "external_url": undefined,
      "feedback_html": undefined,
      "formats": Array [],
      "free_response": undefined,
      "group": undefined,
      "id": 1234,
      "is_completed": undefined,
      "labels": Array [],
      "multiPartGroup": undefined,
      "preview": undefined,
      "response_validation": Object {},
      "spy": Object {},
      "task": Object {},
      "type": undefined,
      "uid": undefined,
    }
  }
  task={StudentTask {}}
/>
`;
